tests/integ/test_reduce_describe_query.py

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import List
+
+import pytest
+
+from snowflake.snowpark._internal.analyzer.expression import Attribute
+from snowflake.snowpark.functions import col
+from snowflake.snowpark.session import (
+    _PYTHON_SNOWPARK_REDUCE_DESCRIBE_QUERY_ENABLED,
+    Session,
+)
+from tests.integ.utils.sql_counter import SqlCounter
+from tests.utils import IS_IN_STORED_PROC
+
+pytestmark = [
+    pytest.mark.skipif(
+        "config.getoption('local_testing_mode', default=False)",
+        reason="Reducing describe queries is not supported in Local Testing",
+    ),
+]
+
+
+def setup(session):
+    is_reduce_describe_query_enabled = session.reduce_describe_query_enabled
+    session.reduce_describe_query_enabled = True
+    yield
+    session.reduce_describe_query_enabled = is_reduce_describe_query_enabled
+
+
+# TODO SNOW-1728988: enable test cases with select after caching attributes on SelectStatement
+# Create from SQL
+create_from_sql_funcs = [
+    lambda session: session.sql("SELECT 1 AS a, 2 AS b"),
+    # lambda session: session.sql("SELECT 1 AS a, 2 AS b").select("b"),
+    # lambda session: session.sql("SELECT 1 AS a, 2 AS b").select("a", lit("2").alias("c")),
+]
+
+# Create from Values
+create_from_values_funcs = [
+    # lambda session: session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"]),
+    # lambda session: session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"]).select("b"),
+    # lambda session: session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"]).select("a", lit("2").alias("c")),
+]
+
+# Create from Table
+create_from_table_funcs = [
+    lambda session: session.create_dataframe(
+        [[1, 2], [3, 4]], schema=["a", "b"]
+    ).cache_result(),
+    # lambda session: session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"]).cache_result().select("b"),
+    # lambda session: session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"]).cache_result().select("a", lit("2").alias("c")),
+]
+
+# Create from SnowflakePlan
+create_from_snowflake_plan_funcs = [
+    lambda session: session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
+    .group_by("a")
+    .count(),
+    lambda session: session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"]).join(
+        session.sql("SELECT 1 AS a, 2 AS b")
+    ),
+    lambda session: session.create_dataframe(
+        [[1, 2], [3, 4]], schema=["a", "b"]
+    ).rename({"b": "c"}),
+    # lambda session: session.range(10).to_df("a")
+]
+
+metadata_no_change_df_ops = [
+    lambda df: df.filter(col("a") > 2),
+    lambda df: df.filter((col("a") - 2) > 2),
+    lambda df: df.sort(col("a").desc()),
+    lambda df: df.sort(-col("a")),
+    lambda df: df.limit(2),
+    # TODO SNOW-1728988: enable this test case (no flatten) after caching attributes on SelectStatement
+    # lambda df: df.sort(col("a").desc()).limit(2).filter(col("a") > 2),
```

Comment:
same

Reply:
sure I can remove them and add later in next PR